frontends/election-manager/src/lib/backends/memory_backend.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  ElectionDefinition,
+  FullElectionExternalTally,
+  Iso8601Timestamp,
+  safeParseElectionDefinition,
+} from '@votingworks/types';
+import { PrintedBallot } from '../../config/types';
+import { ElectionManagerStoreBackend } from './types';
+import { CastVoteRecordFiles } from '../../utils/cast_vote_record_files';
+
+/**
+ * An in-memory backend for ElectionManagerStore. Useful for tests or an
+ * ephemeral session.
+ */
+export class ElectionManagerStoreMemoryBackend
```

Comment:
Why mock at the level of the backend vs the actual storage itself? Seems like we'll get more thorough testing out of it if we mock storage (and then after switching to a server backend, mock fetch)

Reply:
Mainly because, when we do switch to fetching from the backend service, I don't want to have `fetchMock.*` calls all over the place. I'd rather be able to have my own mock that can be more focused. And the `backend` object itself can be heavily unit tested, which I think should provide pretty solid integration assurance.

Reply:
That makes sense. `fetchMock` can definitely be annoying. I built a replacement in Arlo because I couldn't deal with it haha.

I think the downside that I've seen with building these custom mocks is that you then have to ensure that the mock behaves as expected, which requires thinking and/or tests.

Tradeoffs...